### ISSUE TITLE
Question about sWUGGY and sBLIMP

### ISSUE THREAD

**Issue by gallilmaimon** [facebookresearch/spiritlm#14]
Thanks again for publishing this great work!

I have a question about the paper, probably for @tuanh208 :) When calculating the log-likelihood of a speech sample for modelling metrics like sWUGGY, sBLIMP and StoryCloze - do you consider the logits of the speech tokens only (and set the text token logits to -inf)?

**Comment by tuanh208** [comment]
Hi, no I think we compute the log-softmax directly from model logits without setting manually the text tokens logits to -inf

**Comment by gallilmaimon** [comment]
Thanks for the quick reply! One more question, did you normalise the log-likelihood by sequence length as many studies do or use the unnormalised log-likelihood?

I am trying to calculate the swuggy and sblimp metrics with this code and am getting (slightly) worse results than reported and trying to find any discrepancy. Thanks again.

**Comment by tuanh208** [comment]
We did try both (normalise and unnormalised, you can see the comparison in Table 10 in paper) but we reported the normalised in the main tables. 
We also reported the scores obtained from the evaluations on the models trained with our training codebase, and for the release we modified the model checkpoints to make it compatible with Transformers format, so there could be some discrepancies there.
It is also worth nothing that for this release, we used another checkpoint trained on ~175k steps, while in the paper we compared all ablation models with the same 100k steps, but from our side the evaluation scores are roughly the same (there are some metrics with worse scores, some metrics with better scores).

**Comment by Adel-Moumen** [comment]
> We did try both (normalise and unnormalised, you can see the comparison in Table 10 in paper) but we reported the normalised in the main tables. We also reported the scores obtained from the evaluations on the models trained with our training codebase, and for the release we modified the model checkpoints to make it compatible with Transformers format, so there could be some discrepancies there. It is also worth nothing that for this release, we used another checkpoint trained on ~175k steps, while in the paper we compared all ablation models with the same 100k steps, but from our side the evaluation scores are roughly the same (there are some metrics with worse scores, some metrics with better scores).

Hi @tuanh208, 

Hope you are doing well!

Could you please elaborate on why you decided to not release the exact models you used for your paper ? Additionally, do you plan to clarify what dataset you used to train your models ? In your article, you mentioned "We employ open-sourced large-scale speech datasets, totaling 460K hours of speech or 30B speech tokens" without going into the details of what datasets exactly even though they are open source. 

Furthermore, it was surprising to me the first time I read it, but you said in Future work that "SPIRIT-LM models were trained on a large amount of non-English data.". Do you think you will also elaborate a bit more on the quantity of English data that you used ? 

Thanks and its very nice to see a first "large scale" experiment of alternating speech and text (AST) [interleaving] :)

**Comment by tuanh208** [comment]
We found that continue training helps the cross-modal generation a bit better (even if the evaluation metrics are not much improving), so we decided to release it.

For the dataset, unfortunately we couldn't share the details of the components, but most of them are open-sourced, and I do believe that you can achieve comparable results by using open-source datasets (and maybe even better with more recent llms such as llama3 or mistral :D). For the language composition, we started by using a collection of multilingual dataset, with maybe about ~50% of English. We did a small ablation by training the model on English only data and found the results (on English evals) were similar so we decided to keep the non-English part in the training data.

Hope that helps :)

**Comment by Adel-Moumen** [comment]
> For the dataset, unfortunately we couldn't share the details of the components, but most of them are open-sourced, and I do believe that you can achieve comparable results by using open-source datasets (and maybe even better with more recent llms such as llama3 or mistral :D). For the language composition, we started by using a collection of multilingual dataset, with maybe about ~50% of English. We did a small ablation by training the model on English only data and found the results (on English evals) were similar so we decided to keep the non-English part in the training data.

OH! I thought that the datasets you used were fully open-sourced (I think this is what is being said in the paper?).

Thanks anyway for the details :) Yep, it does help. :)

**Comment by gallilmaimon** [comment]
@tuanh208 Thanks for your detailed responses. My results are indeed in the ballpark of the reported swuggy and sblimp results, with the small differences likely explained by a slightly different release version. I did find that setting text tokens logits to -inf improves performance a bit as well. Closing the issue :)
